compiler/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -115,6 +115,13 @@ pub struct Resolver<'a> {
     /// that are captured. We do this in order to create the hidden environment
     /// parameter for the lambda function.
     lambda_stack: Vec<LambdaContext>,
+
+    /// True if we're currently resolving an unconstrained function
+    in_unconstrained_fn: bool,
+
+    /// How many loops we're currently within.
+    /// This increases by 1 at the start of a loop, and decreases by 1 when it ends.
+    nested_loops: u32,
```

Comment:
Should this match `FunctionContext::loops.len()`? If so, would it be useful to store the current `self.nested_loops` value in e.g. `HirStatement::Break`, `HirStatement::Continue` for consistency-checking when we have the `FunctionContext`?

Reply:
I don't know, it feels a bit odd to me to bring this internal detail all the way through the monomorphized IR as well just so that we can assert on it.

Reply:
My current thinking is that although they should match, they also serve different purposes since the resolver is meant to just check "are we currently in a loop?" and issue an error otherwise, where the SSA pass actually tracks the blocks for each loop to break out of. So I'm not sure about making the "are we currently in a loop" check authoritative over the SSA part as well.